account_avatax_oca/models/account_tax.py

```diff
@@ -82,26 +82,39 @@ def compute_all(
             fixed_multiplicator,
         )
         avatax_invoice = self.env.context.get("avatax_invoice")
+        current_aml = False
+        if "current_aml" in self.env.context:
+            current_aml = self.env["account.move.line"].browse(
+                self.env.context.get("current_aml")
+            )
+            if not (
+                current_aml.display_type == "product"
+                and current_aml.account_type != "asset_receivable"
+            ):
+                avatax_invoice = False
         if avatax_invoice:
             # Find the Avatax amount in the invoice Lines
             # Looks up the line for the current product, price_unit, and quantity
             # Note that the price_unit used must consider discount
             base = res["total_excluded"]
             digits = 6
             avatax_amount = None
-            for line in avatax_invoice.invoice_line_ids:
-                price_unit = line.currency_id._convert(
-                    price_unit,
-                    avatax_invoice.company_id.currency_id,
-                    avatax_invoice.company_id,
-                    avatax_invoice.date,
-                )
-                if (
-                    line.product_id == product
-                    and float_compare(line.quantity, quantity, digits) == 0
-                ):
-                    avatax_amount = copysign(line.avatax_amt_line, base)
-                    break
+            if current_aml:
+                avatax_amount = copysign(current_aml.avatax_amt_line, base)
+            else:
+                for line in avatax_invoice.invoice_line_ids:
+                    price_unit = line.currency_id._convert(
+                        price_unit,
+                        avatax_invoice.company_id.currency_id,
+                        avatax_invoice.company_id,
+                        avatax_invoice.date,
+                    )
+                    if (
+                        line.product_id == product
+                        and float_compare(line.quantity, quantity, digits) == 0
+                    ):
+                        avatax_amount = copysign(line.avatax_amt_line, base)
+                        break
```

Comment:
A lot of logic changes here that I don't understand , I'm not comfortable with this.

Reply:
This change is only for adding a process of current aml, if that is not found, should be done standard behavior, you can notice change in showed only for tab adding in else condition